prog/postgres/postgres_nexus.rb

```diff
@@ -0,0 +1,184 @@
+    nap 5 unless vm.strand.label == "wait"
+    vm.sshable.update(host: vm.ephemeral_net4)
+    hop_bootstrap_rhizome
+  end
+
+  label def bootstrap_rhizome
+    register_deadline(:wait, 10 * 60)
+
+    bud Prog::BootstrapRhizome, {"target_folder" => "postgres", "subject_id" => vm.id, "user" => "ubi"}
+    hop_wait_bootstrap_rhizome
+  end
+
+  label def wait_bootstrap_rhizome
+    reap
+    hop_mount_data_disk if leaf?
+    donate
+  end
+
+  label def mount_data_disk
+    case vm.sshable.cmd("common/bin/daemonizer --check format_disk")
+    when "Succeeded"
+      vm.sshable.cmd("sudo mkdir -p /dat")
+      device_path = vm.vm_storage_volumes.find { _1.boot == false }.device_path.shellescape
+
+      vm.sshable.cmd("sudo mount #{device_path} /dat")
+      hop_install_postgres
+    when "Failed", "NotStarted"
+      device_path = vm.vm_storage_volumes.find { _1.boot == false }.device_path.shellescape
+      vm.sshable.cmd("common/bin/daemonizer 'sudo mkfs --type ext4 #{device_path}' format_disk")
+    end
+
+    nap 5
+  end
+
+  label def install_postgres
+    case vm.sshable.cmd("common/bin/daemonizer --check install_postgres")
+    when "Succeeded"
+      server.incr_restart
+      hop_configure
+    when "Failed", "NotStarted"
+      vm.sshable.cmd("common/bin/daemonizer 'sudo postgres/bin/install_postgres' install_postgres")
+    end
+
+    nap 5
+  end
+
+  label def configure
+    case vm.sshable.cmd("common/bin/daemonizer --check configure")
+    when "Succeeded"
+      hop_create_billing_record
+    when "Failed", "NotStarted"
+      configure_hash = server.configure_hash
+      vm.sshable.cmd("common/bin/daemonizer 'sudo postgres/bin/configure' configure", stdin: JSON.generate(configure_hash))
+    end
+
+    nap 5
+  end
+
+  label def create_billing_record
+    BillingRecord.create_with_id(
+      project_id: server.project_id,
+      resource_id: server.id,
+      resource_name: server.server_name,
+      billing_rate_id: BillingRate.from_resource_properties("PostgresCores", "standard", server.location)["id"],
+      amount: vm.cores
+    )
+
+    BillingRecord.create_with_id(
+      project_id: server.project_id,
+      resource_id: server.id,
+      resource_name: server.server_name,
+      billing_rate_id: BillingRate.from_resource_properties("PostgresStorage", "standard", server.location)["id"],
+      amount: server.target_storage_size_gib
+    )
+
+    when_restart_set? do
+      hop_restart
+    end
+
+    hop_wait
+  end
+
+  label def restart
+    decr_restart
+    vm.sshable.cmd("sudo postgres/bin/restart")
+    hop_wait
+  end
+
+  label def wait
+    nap 30
+  end
+
+  label def destroy
+    register_deadline(nil, 5 * 60)
+    reap
+
+    decr_destroy
+
+    if vm
+      vm.sshable&.destroy
+      vm.incr_destroy
+      donate
+    end
+
+    server.dissociate_with_project(server.project)
+    server.destroy
+
+    pop "postgres server is deleted"
+  end
+
+  def server
+    @server ||= PostgresServer[strand.id]
+  end
```

Comment:
You can use `subject_is` helper. If you want to use `server` instead of `postgres_server`, we may need to improve `subject_is` functionality.
```
subject_is: postgres_server
```